Research/lva-node-app/app.js

```diff
@@ -0,0 +1,331 @@
+    });  
+  }
+})
+
+/**
+* this method connects to the IoTHub, and lists the currently running  modules on your device. Ensures connection works properly!
+* Here be dragons :) I double-promise it works as it should!
+*/
+app.post('/connectToIotHub', function (req, res) 
+{
+  setConfigs(req, res).then(() =>
+  {
+    iotHubConnection(req, res).then((response) => 
+    {
+      let modules="";
+      //get modules on device
+      for(let i=0; i<response.responseBody.length; i++)
+      {
+        modules+=response.responseBody[i].moduleId+" ";
+        console.log(response.responseBody[i].moduleId);
+      }
+      //on success
+      res.send("Successfully connected to your IoTHub. Found the following modules on your device: "+modules);
+    }).catch((reason) =>
+    {
+      console.error(reason);
+      res.send("Unable to connect to device. Error response: "+ JSON.parse(reason.responseBody).Message);
+    });    
+  }).catch((error) =>
+  {
+    console.error(error);
+    res.send("Error: "+error);
+  })
+}) 
+
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// defined functions
+
+/**
+* set device ID and IoTHub connection string
+* try to connect to the iot hub to make sure that the passed in values are valid!
+* @returns {Promise<any>} - Promise to resolve/reject based on validity of Device ID and Iot Hub Connection String
+*/
+function setConfigs(req, res)
+{
+  return new Promise((resolve, reject) =>
+  {
+    DEVICE_ID=req.body["device-id"];
+    IOTHUB_CONNECTION_STRING=req.body["iothub-connection-string"];
+    if (DEVICE_ID == "" || IOTHUB_CONNECTION_STRING == "")
+    {
+      DEVICE_ID="";
+      IOTHUB_CONNECTION_STRING="";
+      reject("You must provide both your device ID and your Iot Hub Connection String");
+    }
+    convertIotHubToEventHubsConnectionString(IOTHUB_CONNECTION_STRING).then((eventhubString) =>
+    {
+      IOTHUB_ENDPOINT = eventhubString;
+      resolve();
+    }).catch((error) =>
+    {
+      //reset values on failure
+      console.error(error);
+      DEVICE_ID="";
+      IOTHUB_CONNECTION_STRING="";
+      reject(error.message);
+    });
+  })
+}
+
+
+/** 
+* broadcast used for sending IoT Hub message data in a live stream 
+* @param {any} data - data to send through websocket
+*/
+wss.broadcast = (data) => 
+{
+    wss.clients.forEach((client) => 
+    {
+      if (client.readyState === WebSocket.OPEN) 
+      {
+        try 
+        {
+          client.send(data);
+        } catch (error) 
+        {
+          console.error(error);
+        }
+      }
+    });
+};
+
+/**
+* function that actually broadcasts hub messages to websocket
+* @async
+*/
+async function receiveHubMessages() 
+  {
+      const consumerGroup = "$Default"; // name of the default consumer group
+      eventHubReader = new EventHubReader(IOTHUB_CONNECTION_STRING, consumerGroup);
+      (async () => 
```

Comment:
why are we wrapping an async function in another async and awaiting it? when not do the processing after the await block in the then statement?

Reply:
got rid of async here. Is it ok to leave eventhubreader class methods as async? That way we don't need to do all the .then and .catch-it's cleaner as is but it is a different way of handling things

Reply:
resolved